### PR TITLE
Update kafka documentation

### PR DESCRIPTION
**Pull request by DedunuKarunarathne**
Issue: https://github.com/wso2-enterprise/wso2-mi-internal/issues/700

### CODE REVIEW

File: en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md [hunk 1]
```diff
@@ -24,29 +24,63 @@ Follow these steps to set up the Integration Project and the Connector Exporter
...
 
-2. Specify the API name as `KafkaTransport` and API context as `/publishMessages`. You can go to the source view of the XML configuration file of the API and copy the following configuration (source view).
+    <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-add-connection.png" title="Create Connection" width="800" alt="Create Connection"/></a>
 
+    In the popup window, the following parameters must be provided.
+- Bootstrap Servers - The Kafka brokers listed as host1:port1 and host2:port2.
+- Key Serializer Class - The serializer class for the key that implements the serializer interface.
+- Value Serializer Class - The serializer class for the value that implements the serializer interface.
+
+    <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-connection-configuration.png" title="Connection Configuration" width="800" alt="Connection Configuration"/></a>
```
Comment (SanojPunchihewa) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md:43]: This screenshot should be under Avro Message example

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 2]
```diff
@@ -18,51 +18,180 @@ Follow these steps to set up the Integration Project and the Connector Exporter
...
+6. Next drag and drop the 'publishMessages' operation of the KafkaTransport Connector to the Design View as shown below.
+   <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-avro-example.png" title="Kafka Avro Example" width="800" alt="Kafka Avro Example"/></a>
+
+7. Create a connection from the properties window by clicking on the '+' icon as shown below.
+
+   <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-add-connection.png" title="Create Connection" width="800" alt="Create Connection"/></a>
+
+   In the popup window, provide the following parameters and click **Finish**.
+
+   <table>
```
Comment (SanojPunchihewa) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:92]: This table is not getting rendered properly

File: en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md (hunk 1 above)
Comment (DedunuKarunarathne) [reply en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md]: Fixed with 1889736aed7fbdd77eb868c72dfc10b8f7049056

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md (hunk 2 above)
Comment (DedunuKarunarathne) [reply en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:90]: Fixed with 1889736aed7fbdd77eb868c72dfc10b8f7049056

File: en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md [hunk 3]
```diff
@@ -24,29 +24,62 @@ Follow these steps to set up the Integration Project and the Connector Exporter
...
-                </kafkaTransport.publishMessages configKey="KAFKA_CONNECTION">
+                    <partitionNo>0</partitionNo>
+                </kafkaTransport.publishMessages>
             </inSequence>
             <outSequence/>
             <faultSequence/>
         </resource>
     </api>
     ```
+14. Following is the local entry generated:
```
Comment (SanojPunchihewa) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md:68]:
```suggestion
14. Following is the generated local entry: 
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 4]
```diff
@@ -141,18 +269,9 @@ Sample API configuration when the Confluent Schema Registry is secured with basi
...
                       scope="default"
                       type="STRING"/>
-            <kafkaTransport.init>
-               <name>Sample_Kafka</name>
-               <bootstrapServers>localhost:9092</bootstrapServers>
-               <keySerializerClass>io.confluent.kafka.serializers.KafkaAvroSerializer</keySerializerClass>
-               <valueSerializerClass>io.confluent.kafka.serializers.KafkaAvroSerializer</valueSerializerClass>
-               <schemaRegistryUrl>http://localhost:8081</schemaRegistryUrl>
-               <maxPoolSize>100</maxPoolSize>
-               <basicAuthCredentialsSource>USER_INFO</basicAuthCredentialsSource>
```
Comment (SanojPunchihewa) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:151]: We have described `basicAuth` in the next section. Shall we add this `init` as a local entry for clarity?

Comment (DedunuKarunarathne) [reply en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:151]: Done with d9c38e1ccfe0de6eaf9375ff83d04a3283e6eb67

File: en/docs/reference/connectors/kafka-connector/setting-up-kafka.md [hunk 5]
```diff
@@ -20,9 +20,10 @@ Copy the following additional client libraries to the `<MI_HOME>/lib` directory.
 
 To use the Kafka connector, download and install [Apache Kafka](http://kafka.apache.org/downloads.html). Before you start configuring Kafka you also need the integration runtime and we refer to that location as `<PRODUCT_HOME>`.
 
-> **Note**: The recommended version is Kafka 2.12-2.8.2. For all available versions of Kafka that you can download, see https://kafka.apache.org/downloads. The recommended Java version is 11.
+> **Note**: The Kafka connector has been tested with the following Kafka versions. For all available versions of Kafka that you can download, see https://kafka.apache.org/downloads. The recommended Java version is 11.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/setting-up-kafka.md:23]:
```suggestion
!!!Note
    - The Kafka connector has been tested with the Kafka versions listed below. 
    - To access all downloadable versions of Kafka, see the [Kafka documentation](https://kafka.apache.org/downloads). 
    - The recommended version of Java is 11.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 6]
```diff
@@ -4,65 +4,193 @@ Given below is a sample scenario that demonstrates how to send Apache Avro messa
...
-                </kafkaTransport.publishMessages>
-            </inSequence>
-            <outSequence/>
-            <faultSequence/>
-        </resource>
-    </api>
-    ```
+1. Specify the API name as `KafkaTransport` and API context as `/publishMessages`.
+
+2. To configure the resource click on the API Resource and go to **Properties** view. Select the method POST.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:26]:
```suggestion
2. To configure the resource, click on the API Resource and go to **Properties** view. Select the POST method.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 7]
```diff
@@ -4,65 +4,192 @@ Given below is a sample scenario that demonstrates how to send Apache Avro messa
...
+        <td>json-eval($.key)</td>
+      </tr>
+
+      <tr>
+        <td>topic</td>
+        <td>json-eval($.topic)</td>
+      </tr>
+    </table>
+   
+6. Next drag and drop the 'publishMessages' operation of the KafkaTransport Connector to the Design View as shown below.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:86]:
```suggestion
6. Next drag and drop the `publishMessages` operation of the KafkaTransport Connector to the Design View as shown below.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 8]
```diff
@@ -4,65 +4,192 @@ Given below is a sample scenario that demonstrates how to send Apache Avro messa
...
+   <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-avro-example.png" title="Kafka Avro Example" width="800" alt="Kafka Avro Example"/></a>
+
+7. Create a connection from the properties window by clicking on the '+' icon as shown below.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:89]:
```suggestion
7. Create a connection from the properties window by clicking on the `+` icon as shown below.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 9]
```diff
@@ -4,65 +4,192 @@ Given below is a sample scenario that demonstrates how to send Apache Avro messa
...
+      <tr>
+         <td>Schema Registry URL</td>
+         <td>http://localhost:8081</td>
+      </tr>
+   </table>
+
+    
+    <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-connection-configuration.png" title="Connection Configuration" width="500" alt="Connection Configuration"/></a>
+
+10. After the connection is successfully created, select the created connection as 'Connection' from the drop-down menu in the properties window.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:119]:
```suggestion
10. After the connection is successfully created, select the created connection as `Connection` from the drop-down menu in the properties window.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 10]
```diff
@@ -4,65 +4,192 @@ Given below is a sample scenario that demonstrates how to send Apache Avro messa
...
+
+11. Next, configure the following parameters in the properties window
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:121]:
```suggestion
11. Next, configure the following parameters in the properties window.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md [hunk 11]
```diff
@@ -4,65 +4,192 @@ Given below is a sample scenario that demonstrates how to send Apache Avro messa
...
+                <value>{ctx:value}</value>
+            </kafkaTransport.publishMessages>
+        </inSequence>
+        <outSequence/>
+        <faultSequence/>
+    </resource>
+</api>
+```
+
+13. The following is the generated KAFKA_CONNECTION_1.xml which has been created in local-entries.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-avro-producer-example.md:177]:
```suggestion
13. The following is the generated `KAFKA_CONNECTION_1.xml` which has been created in local-entries.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md [hunk 12]
```diff
@@ -24,29 +24,62 @@ Follow these steps to set up the Integration Project and the Connector Exporter
 
 {!includes/reference/connectors/importing-connector-to-integration-studio.md!}
 
-5. Create a new Kafka connection by selecting a particular operation.
+7. Specify the API name as `KafkaTransport` and API context as `/publishMessages`. 
 
-    <a href="{{base_path}}/assets/img/integrate/connectors/filecon10.png"><img src="{{base_path}}/assets/img/integrate/connectors/filecon10.png" title="working directory" width="800" alt="working directory"/></a>
+8. To configure the resource click on the API Resource and go to **Properties** view. Select the method POST.
 
+9. Next drag and drop the 'publishMessages' operation of the KafkaTransport Connector to the Design View as shown below.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md:31]:
```suggestion
9. Next drag and drop the `publishMessages` operation of the KafkaTransport Connector to the Design View as shown below.
```

File: en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md [hunk 13]
```diff
@@ -24,29 +24,62 @@ Follow these steps to set up the Integration Project and the Connector Exporter
...
-2. Specify the API name as `KafkaTransport` and API context as `/publishMessages`. You can go to the source view of the XML configuration file of the API and copy the following configuration (source view).
+    <a href="{{base_path}}/assets/img/integrate/connectors/kafka"><img src="{{base_path}}/assets/img/integrate/connectors/kafka/kafka-add-connection.png" title="Create Connection" width="800" alt="Create Connection"/></a>
 
+    In the popup window, the following parameters must be provided.
+
+    - Bootstrap Servers - The Kafka brokers listed as host1:port1 and host2:port2.
+    - Key Serializer Class - The serializer class for the key that implements the serializer interface.
+    - Value Serializer Class - The serializer class for the value that implements the serializer interface.
+
+11. After the connection is successfully created, select the created connection as 'Connection' from the drop down menu in the properties window.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/kafka-connector/kafka-connector-producer-example.md:44]:
```suggestion
11. After the connection is successfully created, select the created connection as `Connection` from the drop down menu in the properties window.
```

**Review by SanojPunchihewa — APPROVED** [review]

**Review by DinithiDiaz — APPROVED** [review]

**Pull request merged**